main.js

```diff
@@ -1,144 +1,225 @@
-let main = document.getElementById("mainSec")
-let searchBar = document.getElementById('search')
-let listOfCountries = document.getElementById("options")
-let filter = document.getElementById('filterBar')
-let cards = document.getElementsByClassName("cards-styling")
-console.log(cards)
-let httpReq = new XMLHttpRequest()
-httpReq.open("GET", "data.json");
-httpReq.send()
+// let darkMode = document.querySelector(".themes");
+// let darkModeToggle = document.querySelector(".themes .fa-moon");
+// let navBar = document.querySelector('.nav-bar');
+// let textInDark = document.querySelectorAll('.text-dark');
+// let main = document.getElementById("mainSec");
+// let searchBar = document.getElementById('search');
+// let listOfCountries = document.getElementById("options");
+// let filter = document.getElementById('filterBar');
+// let cards = document.getElementsByClassName("cards-styling");
+let darkMode = document.querySelector(".themes"),
+    darkModeToggle = document.querySelector(".themes .fa-moon"),
+    navBar = document.querySelector(".nav-bar"),
+    textInDark = document.querySelectorAll(".text-dark"),
+    main = document.getElementById("mainSec"),
+    searchBar = document.getElementById("search"),
+    listOfCountries = document.getElementById("options"),
+    filter = document.getElementById("filterBar"),
+    cards = document.getElementsByClassName("cards-styling");
+
+// let httpReq = new XMLHttpRequest()
+// httpReq.open("GET", "data.json");
+// httpReq.send()
+// httpReq.onload = function(){
+//     if(httpReq.status>=200 && httpReq.status <300 && httpReq.readyState == 4 ){
+//         countries = JSON.parse(httpReq.responseText)
+//         // console.log(countries)
+//         getCountries()
+//         fillOptions()
+//         saveTheCurrentTheme()
+//     }
+//     else {
+//         console.error('Failed to load data:', httpReq.statusText);
+//     }
+// }
+
 let countries = [];
-httpReq.onload = function(){
-    if(httpReq.status>=200 && httpReq.status <300 && httpReq.readyState == 4 ){
-        countries = JSON.parse(httpReq.responseText)
-        // console.log(countries)
-        getCountries()
-        fillOptions()
-        saveTheCurrentTheme()
-    }
-    else {
-        console.error('Failed to load data:', httpReq.statusText);
+fetch("data.json", {
+    method: "GET",
+})
+    .then((response) => response.json())
+    .then((resp) => {
+        countries = resp;
+        getCountries();
+        fillOptions();
+        saveTheCurrentTheme();
+    })
+    .catch((e) => {
+        console.error("Failed to load data:", e);
+    });
+
+function saveTheCurrentTheme() {
+    // if(localStorage.getItem('Nav')=="nav-bar-dark" && localStorage.getItem('texts') == "dark-mode-texts") {
+    if (localStorage.getItem("mode") == "dark") {
+        document.body.classList.add("dark");
+        document.body.classList.remove("light");
+        darkModeToggle.classList.replace("fa-regular", "fa-solid");
+

```

Comment:
57 : 69 => -1

this is not an efficient way to create dark mode. 
While all the changes are just classes, we can make one class change and start to make the dark style in the CSS file depending on the class that has been changed.